flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotProvider.java

```diff
@@ -102,31 +93,19 @@ default void start(ComponentMainThreadExecutor mainThreadExecutor) {
 			allocationTimeout);
 	}
 
-	/**
-	 * Allocates a bulk of physical slots. The allocation will be completed
-	 * normally only when all the requests are fulfilled.
-	 *
-	 * @param physicalSlotRequests requests for physical slots
-	 * @param timeout indicating how long it is accepted that the slot requests can be unfulfillable
-	 * @return future of the results of slot requests
-	 */
-	default CompletableFuture<Collection<PhysicalSlotRequest.Result>> allocatePhysicalSlots(
-		Collection<PhysicalSlotRequest> physicalSlotRequests,
-		Time timeout) {
-		throw new UnsupportedOperationException("Not properly implemented.");
-	}
-
 	/**
 	 * Cancels the slot request with the given {@link SlotRequestId} and {@link SlotSharingGroupId}.
 	 *
 	 * @param slotRequestId identifying the slot request to cancel
 	 * @param slotSharingGroupId identifying the slot request to cancel
 	 * @param cause of the cancellation
 	 */
-	void cancelSlotRequest(
-		SlotRequestId slotRequestId,
-		@Nullable SlotSharingGroupId slotSharingGroupId,
-		Throwable cause);
+	default void cancelSlotRequest(
```

Comment:
Looks to me default body is not needed. I tried removing the default body and flink-runtime still compiles.
Correct me if I miss anything.